salt/cli/batch.py

```diff
@@ -312,6 +312,11 @@ def run(self):
                     if self.opts.get('failhard') and data['ret']['retcode'] > 0:
                         failhard = True
 
+                # avoid an exception if the minion does not respond.
+                if data.get("failed") is True:
+                    log.debug('Minion %s failed to respond: data=%s', minion, data)
+                    data = {'ret': 'Minion did not return. [Failed]', 'retcode': salt.defaults.exitcodes.EX_GENERIC}
```

Comment:
I've just noticed that here: https://github.com/saltstack/salt/blob/e81e495b2713336f9908e5f4cc15ba8ebcb67306/salt/cli/batch.py#L311 the structure of 'ret' is different. Shouldn't this be unified?

Reply:
I'm also not sure if completely replacing `data` with a new dictionary is the right thing to do here.

Reply:
@dincamihai 

> the structure of 'ret' is different. Shouldn't this be unified?

The snipped you've mentioned is applied in the only case the executed module returns a `dict` value. In  other cases the return value is cusually `bool`. I the error case this PR handles the return value is absent and it makes it a string. So `retcode` is put in the correct place here.

> I'm also not sure if completely replacing data with a new dictionary is the right thing to do here.

This change is making batch mode work similar to non-batch. In the described case non-batch return is implemented here: https://github.com/saltstack/salt/blob/develop/salt/client/__init__.py#L1580 it just returns a dict containing `out`, `ret` and `retcode` values. Probably it'd be more correct to add `'out': 'no_return'` here too. But overall this PR looks good.

Reply:
> The snipped you've mentioned is applied in the only case the executed module returns a dict value. In other cases the return value is cusually bool. I the error case this PR handles the return value is absent and it makes it a string. So retcode is put in the correct place here.

You are right, but shouldn't we align that exception and have an unified return structure? The structure should not change depending on the executed module.

Reply:
Guys, sorry but I'm still confused. Should I change the return dict? It is still no clear to me what was the decision. =/